src/pytest_fluent/additional_information.py

```diff
@@ -1,31 +1,79 @@
+"""Set additional information function handler."""
+import inspect
 import typing
 
-session_information_callback: typing.Optional[typing.Callable] = None
-test_information_callback: typing.Optional[typing.Callable] = None
+INFORMATION_CALLBACKS: typing.Dict[str, typing.List[typing.Callable]] = {}
+SESSION_STAGE = "pytest_sessionstart"
+TEST_STAGE = "pytest_runtest_logstart"
 
 
-def additional_session_information_callback(fn):
-    global session_information_callback
-    session_information_callback = fn
+def additional_information_callback(stage_name: str):
+    """Set custom informatino callback for any stage.
 
+    Args:
+        stage_name (str): Linked stage name.
+    """
 
-def additional_test_information_callback(fn):
-    global test_information_callback
-    test_information_callback = fn
+    def wrapper(function: typing.Callable):
+        set_additional_information_callback(stage_name, function)
+        return function
 
+    return wrapper
 
-def get_additional_session_information() -> dict:
-    return get_additional_information(session_information_callback)
 
+def additional_session_information_callback(function: typing.Callable):
+    """Set callback for session information.
 
-def get_additional_test_information() -> dict:
-    return get_additional_information(test_information_callback)
+    Args:
+        function (typing.Callable): Callable for test session information.
+    """
+    set_additional_information_callback(SESSION_STAGE, function)
 
 
-def get_additional_information(fn: typing.Optional[typing.Callable]) -> dict:
-    if fn is None:
-        return {}
-    info = fn()
+def additional_test_information_callback(function: typing.Callable):
+    """Set callback for test information.
+
+    Args:
+        function (typing.Callable): Callable for test session information.
+    """
+    set_additional_information_callback(TEST_STAGE, function)
+
+
+def set_additional_information_callback(
+    stage: str,
+    function: typing.Callable,
+) -> None:
+    """Set callable for specific stage.
+
+    Args:
+        stage (str): Stage name.
+        function (typing.Callable): Callback function.
+    """
+    if stage in INFORMATION_CALLBACKS:
+        INFORMATION_CALLBACKS[stage].append(function)
+    else:
+        INFORMATION_CALLBACKS[stage] = [function]
+
+
+def get_additional_information_callback(
+    stage: typing.Optional[str] = None,
+) -> typing.Dict[str, typing.Any]:
+    """Retrieve stage information from callable.
+
+    Args:
+        stage (typing.Optional[str], optional): Stage name. Defaults to None.
+
+    Returns:
+        typing.Dict[str, typing.Any]: Additional information dictionary.
+    """
+    if stage is None:
+        stage = inspect.stack()[1][3]
```

Comment:
Not sure if there is another way to get the pytest hook name, but probably a comment might help to understand what is happening here.